### PR TITLE
week 10

### PR DESCRIPTION
**Pull request by esycrux**

### CODE REVIEW

File: week-10/day-05/BreakWeekPracticeProject/BreakWeekPracticeProject/Controllers/ForumController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        public IActionResult Index(string name)
+        {
+            if (name == null)
+            {
+                return RedirectToAction("login");
+            }
+            else
+            {
+                UserService.currentUser = forumService.FindUserByName(name);
```
Comment (esycrux) [inline week-10/day-05/BreakWeekPracticeProject/BreakWeekPracticeProject/Controllers/ForumController.cs]: I think having some method eg. Login in service and settin user inside will be better approach. So nonone can change oit outside the service.

File: week-10/day-05/BreakWeekPracticeProject/BreakWeekPracticeProject/Services/ForumService.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                context.SaveChanges();
+
+            }
+        }
+
+        public bool UserExists(string name)
+        {
+            using (var context = applicationContext)
+            {
+                if (context.Users.ToList().Find(u => u.UserName==name) == null)
```
Comment (esycrux) [inline week-10/day-05/BreakWeekPracticeProject/BreakWeekPracticeProject/Services/ForumService.cs]: return context.Users.Any(u => u.UserName==name);

File: week-10/day-05/BreakWeekPracticeProject/BreakWeekPracticeProject/Services/UserService.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using BreakWeekPracticeProject.Models;
+
+namespace BreakWeekPracticeProject.Services
+{
+    public static class UserService
+    {
+
+        public static User currentUser;
```
Comment (esycrux) [inline week-10/day-05/BreakWeekPracticeProject/BreakWeekPracticeProject/Services/UserService.cs]: we dont wana have public fields:)

**Comment by pavelnebesky** [comment]
Please review my Forum Java project.

File: week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/controller/ForumController.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class ForumController {
+
+    @Autowired
+    private ForumService forumService;
+
+    @RequestMapping(value="/login")
```
Comment (Kerrialn) [inline week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/controller/ForumController.java]: Here you can just use `@GetMapping(value="/login")` don't need both `@RequestMapping`

File: week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/model/Post.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+@Entity
+public class Post {
+    @Id
+    @GeneratedValue
+    private int id;
+    private String date;
+    private String content;
+    @ManyToOne
+    @JsonIgnore
```
Comment (Kerrialn) [inline week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/model/Post.java]: Good use of Json ignore 👍

File: week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/controller/ForumRestController.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+public class ForumRestController {
```
Comment (Kerrialn) [inline week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/controller/ForumRestController.java]: I would name this class `APIForumController`

File: week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/services/ForumService.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private TopicRepository topicRepository;
+    @Autowired
+    private PostRepository postRepository;
+
+    public User AddAndReturnUser(String username) {
+        User newUser = new User();
```
Comment (Kerrialn) [inline week-12/day-02/forum/src/main/java/com/greenfoxacademy/forum/services/ForumService.java]: You could do this in the model constructor and just pass in the parameter when create the in `User` instance, would make  this the ForumService code cleaner

File: week-12/day-02/forum/src/main/resources/templates/index.html [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        <td> <a th:href="@{'/topic/'+${topic.getId()}}" th:text="${topic.getName()}"> </a>
+        <td th:text="${topic.getPosts().size()}"></td>
+        <td th:text="${topic.getViews()}"></td>
+        <td th:if="${topic.getPosts().size() != 0}" th:text="${topic.getPosts().get(topic.getPosts().size()-1).getUser().getUserName()}"></td>
+        <td th:if="${topic.getPosts().size() == 0}"></td>
+    </tbody>
+</table>
+<br>
+
+<form action="/addtopic" method="post">
```
Comment (Kerrialn) [inline week-12/day-02/forum/src/main/resources/templates/index.html]: remember you can pass in the user object with `th:object="${newTopic}"` and then use `th:field="*{topic}"` on the input

**Pull request merged**